### DISCUSSION TITLE
RocksDB & SSTable interface

### DISCUSSION THREAD

**Discussion by hinxx** [duckdb/duckdb#6612] [Ideas]
I'm thinking on how an interface to RocksDB would look like for DuckDB.

I would be interested to see if the SSTable files could be read and represented as a table (with some upfront knowledge of the key-value "schema" used). One approach would be just ingesting SSTable files (read-only), without starting the RocksDB engine. Otherwise a (read-only) instance of RocksDB could be started. Selecting over key and then "unpacking" the value into column(s) would be of interest initially. In my use case the value is not complex and would fit a 1-4 columns once unpacked. Packing format could be anything from known serialization codes out there or just primitive data type tagging. I could also imagine a JSON object value that could be seen as a "native" JSON file import.

Use case for such idea is when a dedicated application is using RocksDB engine to persist data, but the data handling/analysis is done elsewhere in the organization, with access to immutable SSTable files only.

Having read-write access to the RocksDB engine powered database is not out of the question, but not a primary concern at the moment.

Any thoughts on how this would fit (or not) with the DuckDB in general, and with its pluggable storage concept ?

**Comment by zzzzzzzs** [comment]
Hi, I've been thinking about this recently. Recently, I found that rocksdb writes data very fast, but duckdb is not as fast as rocksdb.
